s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -98,7 +98,7 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
   def request(s3Location: S3Location, rangeOption: Option[ByteRange] = None): Future[HttpResponse] = {
     val downloadRequest = getDownloadRequest(s3Location)
     signAndGet(rangeOption match {
-      case Some(range) => downloadRequest.withHeaders(headers.Range(range))
+      case Some(range) => downloadRequest.withHeaders(downloadRequest.headers :+ headers.Range(range))
```

Comment:
Might be a bit neater to use [addHeader](http://doc.akka.io/api/akka-http/current/akka/http/javadsl/model/HttpRequest.html#addHeader(header:akka.http.javadsl.model.HttpHeader):Self) instead.